src/Compilers/CSharp/Portable/Binder/ImportChain.cs

```diff
@@ -113,11 +113,12 @@ private ImmutableArray<Cci.UsedNamespaceOrType> TranslateImports(Emit.PEModuleBu
                         var assemblyRef = TryGetAssemblyScope(ns, moduleBuilder, diagnostics);
                         usedNamespaces.Add(Cci.UsedNamespaceOrType.CreateNamespace(ns.GetCciAdapter(), assemblyRef, alias));
                     }
-                    else if (target is TypeSymbol { ContainingAssembly.IsLinked: false } typeSymbol)
+                    else if (target is NamedTypeSymbol { ContainingAssembly.IsLinked: false } or not NamedTypeSymbol)
                     {
-                        // We skip alias imports of embedded types to avoid breaking existing code that
-                        // imports types that can't be embedded but doesn't use them anywhere else in the code.
-                        var typeRef = GetTypeReference(typeSymbol, syntax, moduleBuilder, diagnostics);
+                        // We skip alias imports of embedded types to avoid breaking existing code that imports types
+                        // that can't be embedded but doesn't use them anywhere else in the code.  not, this is only
+                        // done for named types.  Other sorts of type symbols (arrays, etc.), are allowed through.
+                        var typeRef = GetTypeReference((TypeSymbol)target, syntax, moduleBuilder, diagnostics);
                         usedNamespaces.Add(Cci.UsedNamespaceOrType.CreateType(typeRef, alias));
```

Comment:
note: there are def open questions on what the right behavior shoudl be here.

First, i'll note that *prior* to any of my changes here, this code was already potentially problematic wrt to instantiated named types (like `List<EmbeddedType>`).  We would not skip those since `List<>` itself was not-linked.   Unclear if that was desirable or not.

If that is not desirable, this needs to walk the entire symbol (including type arguments) to see if there are any embedded types.

If the above *is* intentional, and only the top-level construct is checked, then the new code follows the same logic.  Namely that as all other types themselves are not-linked, though internally they may have a linked type (for example `EmbeddedType[]`), then it's ok to translate them along.

So, this code keeps similar semantics as before.  We just need to decide if we want those semantics, or if we want to change things such that any usage of an embedded type would prevent translation. 

tagging @AlekseyTs in case he has thoughts on this.